src/components/RoomSelection/RoomSelectionCards.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from "react";
+import {  Box, Typography, Card } from "@mui/material";
+import COLORS from "../../constants/CustomColors";
+import { iRoomCards } from "../../interfaces/interfaces";
+import { useNavigate } from "react-router-dom";
+export const RoomSelectionCards = ({
+    avaibility,
+    description,
+    id,
+    name,
+    capacity,
+}: iRoomCards) => {
+    const navigate = useNavigate();
+    const [color, setColor] = useState<string>();
+    const handleClick = (id: number) => {
+        navigate(`/rooms/${id}`);
+    };
+    useEffect(() => {
+        if (avaibility === "free") {
+            setColor(COLORS.SUCCESS);
+        } else {
+            if (avaibility === "booked") {
+                setColor(COLORS.ERROR);
+            } else {
+                setColor(COLORS.WARNING);
+            }
+        }
```

Comment:
instead of writing that many if else here you can have a mapping object with keys = status and value = enum and after that just access the value 
`const obj = {
    "free": COLORS.SUCCESS,
    "booked": COLORS.ERROR,
    "some_value": COLORS.WARNING
}`

and now you can write obj[availability]